server/pbench/bin/index-pbench

```diff
@@ -270,14 +279,23 @@ class PbenchData(object):
             # timestamp.  If the new absolute timestamp is beyond the end of
             # the run, it is likely that the timestamp is not relative so we
             # raise an error.
-            d = timedelta(0, 0, orig_ts * 1000)
-            ts = self.ptb.start_run_ts + d
-            if ts > self.ptb.end_run_ts:
+            try:
+                d = timedelta(0, 0, orig_ts_float * 1000)
+            except Exception as e:
+                raise BadDate("{:f} ({!r}) is not a proper float in"
+                        " milliseconds since the epoch: {}".format(
+                            orig_ts_float, orig_ts, e))
+            newts = self.ptb.start_run_ts + d
+            if newts > self.ptb.end_run_ts:
                 self.counters['ts_before_start_run_ts'] += 1
-                raise BadDate()
+                raise BadDate("{} ({!r}) is before the start of the"
```

Comment:
I thought I understood this logic last time but I'm still unsure about it. Maybe we can chat a bit and draw some diagrams.